src/check_result.jl

```diff
@@ -33,6 +33,12 @@ check_equal(x::Zero, y::Zero; kwargs...) = @test true
 check_equal(x::DoesNotExist, y::Nothing; kwargs...) = @test true
 check_equal(x::Nothing, y::DoesNotExist; kwargs...) = @test true
 
+check_equal(::ChainRulesCore.NotImplemented, x; kwargs...) = @test_broken false
```

Comment:
```suggestion
# Checking equality with `NotImplemented` reports `@test_broken` since the derivative has intentionalyl
# not yet been implemented
check_equal(::ChainRulesCore.NotImplemented, x; kwargs...) = @test_broken false
```